components/shared/net/request.rs

```diff
@@ -22,8 +22,8 @@ use crate::{ReferrerPolicy, ResourceTimingType};
 /// An id to differeniate one network request from another.
 pub struct RequestId(usize);
 
-impl RequestId {
-    pub fn new() -> Self {
+impl Default for RequestId {
+    fn default() -> Self {
```

Comment:
`default()` doesn't quite capture what this is doing. Instead please call this `next()`:

```suggestion
impl RequestId {
    fn next() -> Self {
```

Reply:
Okay Sir, let me do that.

Reply:
@mrobinson when ```fn next()``` is used ```build -d``` reports errors ``` RequestId` is not an iterator``` 
Should we rename it to something else?

Reply:
`next_id` sounds great. Thanks for checking!

Reply:
> `next_id` sounds great. Thanks for checking!

@mrobinson, there appears to still be an issue  with renaming it to anything else but ```default``` unless ```fn default``` is implemented in some way. Let me share a screenshot. 

![Screenshot from 2024-10-17 18-31-11](https://github.com/user-attachments/assets/01f8968a-f9b6-41da-bfd4-58235888be03)

Reply:
you need to remove the `impl Default for RequestId` block and use `impl RequestId` instead.

Reply:
> you need to remove the `impl Default for RequestId` block and use `impl RequestId` instead.

That is the warning we are trying to work on, the only other option would be to ```allow``` it

Reply:
The warning is triggered only because the method is currently named `new`. It will disappear if we renamed `new` to `next_id`, so there is no need to `allow` it or have `impl Default`.

Reply:
> The warning is triggered only because the method is currently named `new`. It will disappear if we renamed `new` to `next_id`, so there is no need to `allow` it or have `impl Default`.

You are right, looking at the suggestion provided by @mrobinson ```impl Default``` is eliminated. 
Thank you @mukilan